File: Magic Triplets - GFG/magic-triplets.py
#User function Template for python3
from sortedcontainers import SortedList
class Solution:
	def countTriplets(self, nums):
		front=SortedList()
		rare=SortedList()
		front.add(nums[0])
		for i in range(1,len(nums)):
		    rare.add(nums[i])
		i=1
		ans=0
		while(i<len(nums)-1):
		    rare.discard(nums[i])
		    t1=front.bisect_left(nums[i])
		    r1=0
		    if(t1==len(front)):
		        r1=len(front)
		    else:
		        r1=t1
		    t2=rare.bisect_right(nums[i])
		    r2=len(rare)-t2
		    ans+=(r1*r2)
		    front.add(nums[i])
		    i+=1
		return ans


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
	T=int(input())
	for i in range(T):
		n = int(input())
		nums = list(map(int,input().split()))
		ob = Solution()
		ans = ob.countTriplets(nums)
		print(ans)

# } Driver Code Ends

File: Magic Triplets - GFG/magic-triplets.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution{
	public:
	int countTriplets(vector<int>nums){
	    int ans=0;
	    for(int i=1;i<nums.size()-1;i++){
	        int j=0;
	        int r1=0;
	        while(j<i){
	            if(nums[j]<nums[i])
	                r1+=1;
	            j+=1;
	        }
	        j=i+1;
	        int r2=0;
	        while(j<nums.size()){
	            if(nums[j]>nums[i])
	               r2+=1;
                j+=1;
	        }
            ans+=(r1*r2);
	    }
	    return ans;
	}
};

//{ Driver Code Starts.
int main(){
	int tc;
	cin >> tc;
	while(tc--){
		int n;
		cin >> n;
		vector<int>nums(n);
		for(int i = 0; i < n; i++)cin >> nums[i];
		Solution ob;
		int ans = ob.countTriplets(nums);
		cout << ans << "\n";
	}  
	return 0;
}
// } Driver Code Ends